File: samples/erlang/07env/env.erl
#!/usr/bin/env escript
%% -*- erlang -*-
%%! -smp enable -sname factorial -mnesia debug verbose
%%
-mode(compile).

%% getenv

print_all([H | T]) ->
    io:format("~s~n", [H]),
    print_all(T);
print_all([]) ->
    [].


main([]) ->
    print_all(os:getenv());
main(_) ->
    io:format(standard_error, "Error: Usage env.erl~n", []),
    halt(1).

File: samples/java/07env/env.java
import java.util.Map;

public class env {

    public static void main(String[] args) {
        Map<String, String> env = System.getenv();
        for (String var : env.keySet()) {
            System.out.format("%s=%s\n", var, env.get(var));
        }
    }
}
